io.openems.edge.io.shelly/src/io/openems/edge/io/shelly/shelly25/IoShelly25Impl.java

```diff
@@ -100,68 +116,86 @@ public void handleEvent(Event event) {
 		}
 
 		switch (event.getTopic()) {
-		case EdgeEventConstants.TOPIC_CYCLE_BEFORE_PROCESS_IMAGE:
-			this.eventBeforeProcessImage();
-			break;
-
-		case EdgeEventConstants.TOPIC_CYCLE_EXECUTE_WRITE:
-			this.eventExecuteWrite();
-			break;
+		case EdgeEventConstants.TOPIC_CYCLE_EXECUTE_WRITE //
+			-> this.eventExecuteWrite();
 		}
 	}
 
 	/**
 	 * Execute on Cycle Event "Before Process Image".
+	 * 
+	 * @param result The JSON element containing the result of the HTTP request.
+	 * @param error  The throwable error, if any occurred during the HTTP request.
+	 * @throws OpenemsNamedException if the processing of the HTTP result fails or
+	 *                               communication with the slave device is
+	 *                               unsuccessful.
 	 */
-	private void eventBeforeProcessImage() {
-		Boolean relay1IsOn;
-		Boolean relay2IsOn;
-		try {
-			var json = this.shellyApi.getStatus();
-			var relays = JsonUtils.getAsJsonArray(json, "relays");
-			var relay1 = JsonUtils.getAsJsonObject(relays.get(0));
-			relay1IsOn = JsonUtils.getAsBoolean(relay1, "ison");
-			var relay2 = JsonUtils.getAsJsonObject(relays.get(1));
-			relay2IsOn = JsonUtils.getAsBoolean(relay2, "ison");
+	private void processHttpResult(JsonElement result, Throwable error) {
+		this._setSlaveCommunicationFailed(result == null);
+		Boolean relay1 = null;
+		Boolean overtemp1 = null;
+		Boolean overpower1 = null;
+		Boolean relay2 = null;
+		Boolean overtemp2 = null;
+		Boolean overpower2 = null;
 
+		try {
+			JsonElement jsonElement = JsonUtils.getAsJsonElement(result);
+			final var relays = JsonUtils.getAsJsonArray(jsonElement, "relays");
+			for (int i = 0; i < relays.size(); i++) {
+				final var relay = JsonUtils.getAsJsonObject(relays.get(i));
+				final var relayIson = JsonUtils.getAsBoolean(relay, "ison");
+				final var relayOverpower = JsonUtils.getAsBoolean(relay, "overpower");
+				final var relayOvertemp = JsonUtils.getAsBoolean(relay, "overtemperature");
+
+				if (i == 0) {
+					relay1 = relayIson;
+					overtemp1 = relayOvertemp;
+					overpower1 = relayOverpower;
+				} else if (i == 1) {
```

Comment:
This logic assumes, that there `relays.size() == 2`. There are multiple ways to do it better, e.g. 
- `for (var i : new int[] { 0, 1 }) {`
- `for (int i = 0; i < 2; i++) {`
- add a `record` with relay, overtemp and overpower and call a static `parseRelay()` method twice